patch/MultiDelay/MultiDelay.cpp

```diff
@@ -0,0 +1,138 @@
+#include "daisysp.h"
+#include "daisy_patch.h"
+#include <string>
+
+#define MAX_DELAY static_cast<size_t>(48000 * 1.f)
+
+using namespace daisy;
+using namespace daisysp;
+
+DaisyPatch patch;
+
+DelayLine <float, MAX_DELAY> DSY_SDRAM_BSS delMems[3]; 
+
+struct delay
+{
+    DelayLine <float, MAX_DELAY>  *del;
+    float currentDelay;
+    float delayTarget;
+};
+
+delay delays[3];
+Parameter params[4];
+
+float feedback;
+int drywet;
+
+void ProcessControls();
+
+float UpdateDelay(delay &delay, float in)
+{
+    //set delay times
+    fonepole(delay.currentDelay, delay.delayTarget, .004f);
+    delay.del->SetDelay(delay.currentDelay);
+    
+    float read = delay.del->Read();
+    delay.del->Write((feedback * read) + (1 - feedback) * in);
+
+    return read;
+}
+
+static void AudioCallback(float **in, float **out, size_t size)
+{
+    ProcessControls();
+    
+    for (size_t i = 0; i < size; i ++)
+    {
+	float mix = 0;
+
+	//update delayline with feedback
+	for (int d = 0; d < 3; d++)
+	{
+	    mix += UpdateDelay(delays[d], in[0][i]);
+	}
+
+	//apply drywet and attenuate
+	float fdrywet = (float)drywet / 100.f;
+	mix = ((fdrywet * mix * .3f) + ((1.0f - fdrywet) * in[0][i]));
+
+	for (size_t chn = 0; chn < 4; chn++)
+        {   
+	    out[chn][i] = mix;
+        }
```

Comment:
just a fun idea, since there are some other changes, and its just a copy for each output right now. The first three outputs could be the individual delay lines (for patching to things with panning), and the fourth output could be the mix.

Reply:
So this was implemented, but there's a bug still wherein the other delays can't be heard, except for a zipping effect when their delay times are changed. I.e. On output 1 you can't hear delay 2, except for when you change delay 2's time you can hear that movement somehow.